server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/period/PeriodsHolder.java

```diff
@@ -53,7 +58,23 @@
    * @throws IllegalStateException if the periods haven't been initialized
    * @throws IllegalStateException if there is no period for the specified index (see {@link #hasPeriod(int)})
    * @throws IndexOutOfBoundsException if i is either &lt; 1 or &gt; 5
+   * @deprecated replaced by {@link #getPeriod()}
    */
+  @Deprecated
   Period getPeriod(int i);
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)